python/ray/experimental/state.py

```diff
@@ -419,17 +422,25 @@ def task_profiles(self, start=None, end=None, num_tasks=None, fwd=True):
         task_info = dict()
         event_log_sets = self.redis_client.keys("event_log*")
 
+        if num_tasks is None:
```

Comment:
This code should not be part of the GlobalState API and rather num_tasks should be passed in from the webui

Reply:
Done